### PR TITLE
MAINT: Bump up deps with tox

### PR DESCRIPTION
**Pull request by MridulS**
I had to add a upper pin to numpy to make `tox -e deps` resolve. Numba has an upper pin for numpy.

### CODE REVIEW

File: requirements/docs.txt [hunk 1]
```diff
@@ -139,41 +139,41 @@ snowballstemmer==2.2.0
     # via sphinx
 sortedcontainers==2.4.0
     # via hypothesis
-soupsieve==2.5
+soupsieve==2.6
     # via beautifulsoup4
-sphinx==7.3.7
+sphinx==8.0.2
```
Comment (jl-wynen) [inline requirements/docs.txt:144]: Please check that the docs are ok. Including figures, theme-specifics, etc.

Comment (MridulS) [reply requirements/docs.txt:144]: The build locally looks fine, nothing sticks out.

File: requirements/docs.txt [hunk 2]
```diff
@@ -107,29 +107,29 @@ nbformat==5.10.4
...
+nbsphinx==0.9.5
     # via -r docs.in
 pandocfilters==1.5.1
     # via nbconvert
 pillow==10.4.0
     # via
     #   ipympl
     #   matplotlib
-plopp==24.6.0
+plopp==24.9.0
```
Comment (jl-wynen) [inline requirements/docs.txt:118]: 3d plots are broken. See https://github.com/scipp/scippneutron/pull/543#issuecomment-2349060835 for context and @nvaytet's comment for a solution.

E.g., in `user-guide/coordinate-transformations.html`

**Review by jl-wynen — CHANGES_REQUESTED** [review]
I checked the docs generated by CI. They look fine except for one issue.

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**